tests/test_gridsearch.py

```diff
@@ -131,9 +131,8 @@ def test_grid_search_no_score(self):
             grid_search.score(X, y), grid_search_no_score.score(X, y))
 
         # giving no scoring function raises an error
-        grid_search_no_score = TuneGridSearchCV(clf_no_score, {"C": Cs})
         with self.assertRaises(TypeError) as exc:
-            grid_search_no_score.fit([[1]])
+            grid_search_no_score = TuneGridSearchCV(clf_no_score, {"C": Cs})
```

Comment:
In the first pr for multimetric scoring, we wanted to adhere to the sklearn behavior so that this errors when the user calls fit. I'm aware that it's now necessary to do these checks in init because the metric name is used in the `TuneSearchCV` subclass. Do we think this different behavior is okay? @richardliaw @anthonyhsyu

Reply:
I think it's OK, makes sense to raise earlier!